jetty-server/src/main/java/org/eclipse/jetty/server/HostHeaderCustomizer.java

```diff
@@ -57,15 +60,26 @@ public HostHeaderCustomizer(String serverName)
      */
     public HostHeaderCustomizer(String serverName, int serverPort)
     {
-        this.serverName = Objects.requireNonNull(serverName);
+        this.serverName = serverName;
         this.serverPort = serverPort;
     }
 
     @Override
     public void customize(Connector connector, HttpConfiguration channelConfig, Request request)
     {
-        if (request.getHeader("Host") == null)
-            // TODO set the field as well?
-            request.setHttpURI(HttpURI.build(request.getHttpURI()).host(serverName).port(serverPort));
+        if (request.getHttpVersion() != HttpVersion.HTTP_1_1 && !request.getHttpFields().contains(HttpHeader.HOST))
+        {
+            String host = serverName == null ? request.getServerName() : serverName;
+            int port = HttpScheme.normalizePort(request.getScheme(), serverPort == 0 ? request.getServerPort() : serverPort);
+
+            if (serverName != null || serverPort > 0)
```

Comment:
Don't you want to setHttpURI() if the `port` is normalized in a way that's different then `serverPort` too?

Reply:
My thinking here is that if the serverPort is set, then we will set the port, but will used the normalised version of it.
So if they set port 80, but the real port was 8888 we would set the port to 0 not 80